test-network-nano-bash/README.md

```diff
@@ -0,0 +1,78 @@
+# Test network - Nano bash
+
+Test network Nano bash provides a set of minimal bash scripts to run a Fabric network on your local machine.
+The network is functionally equivalent to the docker-based Test Network, you can therefore run all the tutorials and samples that target the Test Network.
+The Fabric release binaries are utilized rather than using docker containers to avoid all unnecessary layers. Only the chaincode and chaincode builder runs in a docker container behind the scenes.
+Using the Fabric binaries also makes it simple for Fabric developers to iteratively and quickly modify Fabric code and test a Fabric network as a user.
+
+As the name `nano` implies, the scripts provide the smallest minimal setup possible for a Fabric network while still offering a multi-node TLS-enabled network:
+- Minimal set of dependencies
+- Minimal requirements on Fabric version (any v2.x orderer and peer nodes are supported)
+- Minimal set of environment variable overrides of the default orderer orderer.yaml and peer core.yaml configurations
+- Minimal scripting with minimal set of reference commands to get a Fabric network up and running
+- Minimal channel configuration for an orderer organization (3 ordering nodes) and two peer organizations (with two peers each)
+- Minimal endorsement policy to allow a single organization to approve and commit a chaincode
+
+# Prereqs
+
+- Follow the Fabric documentation for the [Prereqs](https://hyperledger-fabric.readthedocs.io/en/latest/prereqs.html)
+- Follow the Fabric documentation for [downloading the Fabric samples and binaries](https://hyperledger-fabric.readthedocs.io/en/latest/install.html). You can skip the docker image downloads by using `curl -sSL https://bit.ly/2ysbOFE | bash -s -- -d`
+
+# Instructions for starting network
+
+Open terminal windows for 3 ordering nodes, 4 peer nodes, and 4 peer admins as seen in the following terminal setup.
+Note, you can start with two ordering nodes and single peer node and single peer admin terminal if you would like to keep things even more minimal (two ordering nodes are required to achieve consensus (2 of 3), while a single peer can be utilized since the endorsement policy is set as any single organization).
+![Terminal setup](terminal_setup.png)
+
+
+- cd to the `test-network-nano-bash` directory in each terminal window
+- In the first orderer terminal, run `./generate_artifacts.sh` to generate crypto material (calls cryptogen) and system and application channel genesis block and configuration transactions (calls configtxgen)
+- In the three orderer terminals, run `./orderer1.sh`, `./orderer2.sh`, `./orderer3.sh` respectively
+- In the four peer terminals, run `./peer1.sh`, `./peer2.sh`, `./peer3.sh`, `./peer4.sh` respectively
+- In the four peer admin terminals, run `. peer1admin.sh`, `. peer2admin.sh`, `. peer3admin.sh`, `. peer4admin.sh` respectively
```

Comment:
It's a neat idea to be able to set the environment like this - however admit that I've 'gone off' this approach.. the reason being that different shells, handle this all a bit differently.

The recent scripts I've done for examples have opted for the approach of printing out the `export x=y` settings that are needed.. The user can then copy and paste them to the shell.  The only really reliable way I found of doing this.

Reply:
Another option is to have a set of `.env` files for each peer... if these just list the `name=value` settings for each variable. then in each shell you can configure the environment variables with a single line..

```
export $(grep -v '^#' peer1admin.env | xargs)
```

Reply:
I've intentionally made the scripts so simple that the commands from the scripts can be copy and pasted and run on their own. So instead of adding more intelligence, I prefer to keep them as simple as possible. I've updated the README to address the concern:

"The following instructions will have you run simple bash scripts that set environment variable overrides for a component and then runs the component.
The scripts contain only simple single-line commands so that they are easy to read and understand.
If you have trouble running bash scripts in your environment, you can just as easily copy and paste the individual commands from the script files instead of running the script files."